Src/BootCamp.Chapter/Lesson3.cs

```diff
@@ -0,0 +1,138 @@
+        }
+        public static float GetFloatInput(string prompt)
+        {
+            Console.WriteLine(prompt);
+            string value = Console.ReadLine();
+
+            if (Empty(value))
+            {
+                return 0;
+            }
+
+            if (float.TryParse(value, out float parsedfloat))
+            {
+                return parsedfloat;
+            }
+
+            //default case
+            BadInput(value);
+            return -1;
+
+        }
+        // Input function
+        private static void GatherData(out string firstName, out string lastName, out int age, out float weight, out float height)
+        {
+            firstName = GetStringInput("Hello, What is your First name?");
+            lastName = GetStringInput("Hello, What is your Last name?");
+
+            age = GetIntInput("How old are you?");
+            weight = GetFloatInput("How much do you weigh in kilograms?");
+            height = GetFloatInput("How tall are you?") / 100;
+            if (height <= 0)
+            {
+                Console.WriteLine("Press [Enter] to continue");
+                Console.ReadKey();
+            }
+        }
+        // Output function
+        public static void DisplayOutput(string first, string last, int age, float weight, float height)
+        {
+            Console.Clear();
+            Console.WriteLine(first + " " + last + " is " + age + " years old; his weight is " + weight + " kg and his height is " + (height * 100) + " cm.");
+            Console.WriteLine(first + "'s Body Mass Index is " + CalcBmi(weight, height) + ".");
+            Console.WriteLine("\n\n\n\n\nPress Enter");
+            Console.ReadLine();
+        }
+        //Calculate BMI
+        public static float CalcBmi(float weight, float height)
+        {
+            if (weight > 0 && height <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                Console.WriteLine("Height cannot be equal or less than zero, but was " + height + ".");
+
+                return -1F;
+            }
+            
+            if (height < 0 || weight<=0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+
+                if (weight <= 0)
+                {
+                    Console.WriteLine("Weight cannot be equal or less than zero, but was " + weight + ".");
+                }
+
+                if (height <= 0 && weight >= 0)
+                {
+                    Console.WriteLine("Height cannot be less than zero, but was " + height + ".");
+                }
+                return -1F;
+            }
+
+            return weight / (height * height);
+        }
+        //outputs error message if given invalid input
+        public static void BadInput(string value)
+        {
+            Console.Write("\"" + value + "\" is not a valid number.");
+        }
+        //returns true on an empty input
+        public static bool Empty(string value)
```

Comment:
Boolean functions (and variables) should express questions that they answer on.
So for example here you would want it to be at least IsEmpty(string value).

Reply:
I would quibble on this one, but ok, done.

Reply:
Feel free to do so!
Sometimes you need to have a conversation about such a thing to understand it right.

I will try to explain it that way:
Assume your project grows and grows and grows and you maybe have not a function like Empty but something less obvious. Now you (or maybe a 3rd party user) want(s) to look up a certain functionality.
By using HasX, IsX, etc. you can easily find such functions with a proper IDE like VS is. Also if you are used to that, you probably will not find Empty anymore and end up writing it again, maybe at some other place in your code.

I hope this makes my intentions behind that much more clear. If not - just ask again! :)